File: backend/app/api/v1/health.py
"""Health check router."""

from fastapi import APIRouter
from pydantic import BaseModel

from app.core.config import settings

router = APIRouter(tags=["health"])


class HealthResponse(BaseModel):
    """Health check response model."""

    status: str
    application: str
    version: str


@router.get("/health", response_model=HealthResponse)
async def health_check() -> HealthResponse:
    """Health check endpoint.

    Returns:
        Health status response
    """
    return HealthResponse(
        status="healthy",
        application=settings.app_name,
        version=settings.version,
    )

File: frontend/src/services/health.ts
/**
 * Health check service
 */
import api from './api';

export interface HealthResponse {
  status: string;
  application: string;
  version: string;
}

export const healthService = {
  checkHealth: async (): Promise<HealthResponse> => {
    const response = await api.get<HealthResponse>('/health');
    return response.data;
  },
};
